### ISSUE TITLE
Combobox with rxjs resetting itself

### ISSUE THREAD

**Issue by rdnscr** [telerik/kendo-angular#864]
We are using the latest version (1.1.3) of Kendo Dropdowns. In there we have the combobox which we use directly witha remoteservice which is doing the filtering. To enable a proper behavior we register in the ng after view init on filter change, to apply our logic:
```
  ngAfterViewInit() {
    this.list.filterChange.asObservable()
        .filter((value: string) => value.length >= this.minCharacters)
        .delay(300)
        .switchMap((value: string) => {
          this.list.loading = true;
          return this.filter(value);
        })
        .subscribe((x: any[]) => {
          this.filteredItems = x.splice(0, this.maxDisplayedItems);
          this.list.loading = false;
          this.cd.markForCheck();
        });

  }
```
This code works in generally perfect. However if i use a delay or a debounceTime there are cases in which the combobox is resetting itself if i do multiple searches (entering searchvalue, waitng, changing search value and so on).

From my point of view it looks like a race condition in the setState function of the combobox:
```
    ComboBoxComponent.prototype.setState = function (value) {
        var primitiveValue = util_1.getter(value, this.valueField, true);
        var valueCandidate;
        var textCandidate;
        var dataItemCandidate;
        if (!this.data.length) {
            if (this._filtering) {
                this.selectionService.resetSelection([-1]);
                return;
            }
            dataItemCandidate = this.valueField && !this.valuePrimitive ? value : undefined;
            valueCandidate = value;
            textCandidate = this.allowCustom ? value : "";
        }
        else {
            if (this._filtering) {
                textCandidate = this.text;
            }
            else {
                var result = util_1.resolveValue({
                    data: this.data,
                    value: primitiveValue,
                    valueField: this.valueField
                });
                valueCandidate = this.valuePrimitive ? primitiveValue : value;
                if (result.dataItem) {
                    this.selectionService.resetSelection(result.selected);
                    textCandidate = result.dataItem;
                    dataItemCandidate = result.dataItem;
                }
                else {
                    this.selectionService.resetSelection([]);
                    this.selectionService.focused = 0;
                    if (this.allowCustom) {
                        textCandidate = value;
                        dataItemCandidate = !this.valuePrimitive ? value : undefined;
                    }
                    else {
                        textCandidate = "";
                        valueCandidate = undefined;
                        dataItemCandidate = undefined;
                    }
                }
            }
        }
        this._filtering = false;
        this.text = textCandidate;
        this.value = valueCandidate;
        this.dataItem = dataItemCandidate;
        this._previousValue = valueCandidate;
    };
```
In the case the combobox is resetting itself the issue is this line 

> this.text = textCandidate;

I don't undestand how this race condition is happening. I assume there is an issue when the comobox is triggering the filter change event or handling it response? As mentioned if no delay/debounceTime is used the issue seems not to occure.

Any help would be appreciated. Since you have no example with dropdowns using remote data i can not provide any plunkr out of the box showing the issue.

**Comment by tapopov** [comment]
Hi @rdnscr 
What you report seem to be the same issue described in https://github.com/telerik/kendo-angular/issues/844. We're already working on it and a fix should be available soon.

**Comment by siyka-marinova** [comment]
Fix for this bug is now available in the latest dev. version of the kendo-angular-dropdowns package: @progress/kendo-angular-dropdowns@1.2.0-dev.201709010832. The official package will be released soon.

**Comment by tsvetomir** [comment]
Fix shipped in `kendo-angular-dropdowns@1.2.0`
